utils/rigid_origami_utils.py

```diff
@@ -0,0 +1,513 @@
+                v1 = bm_e.verts[0].index
+                if v0 in edge and v1 in edge:
+                    bm_e_index = bm_e.index
+                    break
+        return 
+    
+    # Get the object face index from a bmesh face
+    def bm_to_obj_face_index(self, bm_face):
+        v_indices = [v.index for v in bm_face.verts]
+        for i, f in enumerate(self.faces):
+            if set(f) == set(v_indices):
+                return i
+        raise ValueError("Created BMesh faces is wrong")
+        
+# === Crease Lines class ===
+class CreaseLines:
+
+    # Constructor
+    def __init__(self, obj, fold_edge_indices, fold_edge_angles, folding):
+        
+        # Collect edges inside of a mesh
+        edge_indices = [obj.bm_to_obj_edge_index(obj.bm.edges[i]) \
+                        for i, e in enumerate(obj.bm.edges) \
+                        if not e.is_boundary]
+        self.edges = [tuple(obj.edges[i]) for i in edge_indices]
+
+        # Initialize edge angles with the current angles
+        # (round function is needed to calculate pseudo inverse correctly later)
+        self.angles = [round(obj.bm.edges[i].calc_face_angle_signed(0.0), 5) \
+                        for i, e in enumerate(obj.bm.edges) \
+                        if not e.is_boundary]
+                        
+        final_angles = [round(fold_edge_angles[fold_edge_indices.index(e)], 5) \
+                                if (fold_edge_indices.count(e) > 0 \
+                                and fold_edge_indices.index(e) < len(fold_edge_angles)) \
+                                else self.angles[i] \
+                                for i, e in enumerate(edge_indices)]
+        diffs = [final - angle \
+                    for final, angle in zip(final_angles, self.angles)]
+
+        # Target angles multiplyed with folding ratio
+        self.target_angles = [angle + (diff * folding) for angle, diff \
+                                in zip(self.angles, diffs)]
+        
+        # Delta angles to be calculated in this node
+        self.delta_angles = [0.0] * len(self.angles)
+
+# === Inside Vertex class ===
+class InsideVertex:
+    # Vertex indices inside of the paper
+    indices = []
+    
+    # Constructor
+    def __init__(self, vertex, v_index, edges, \
+                    crease_indices, thetas, rhos, drhos):
+        self.vertex = vertex
+        self.v_index = v_index
+        self.edges = edges
+        self.crease_indices = crease_indices
+        self.thetas = thetas
+        self.init_rhos = [rho for rho in rhos]
+        self.rhos = rhos
+        self.drhos = drhos
+        
+    # Function to generate inside vertex objects
+    @classmethod
+    def GenerateInsideVertices(cls, obj, crease_lines):
+        
+        # Create vertex indices inside of the source paper
+        num_verts = len(obj.verts)
+        indices = [v.index for v in obj.bm.verts if not v.is_boundary]
+        InsideVertex.indices = indices
```

Comment:
```suggestion
        cls.indices = indices
```
It has matter in some cases. I think it would be more correct to use `cls` object.

Reply:
Thank you. I'll check whether any other class objects are used in the same way.